File: CODIGO 139/9.js
function solveBeamBalanceAndMasses(test_cases) {
    const masses = [1, 3, 9, 27, 81];
    for (const test_case of test_cases) {
      const power_of_3 = 0;
      while (test_case >= 81) {
        power_of_3++;
        test_case -= 81;
      }
      masses.push(3 ** power_of_3);
      test_case -= 3 ** power_of_3;
    }
    return masses;
  }
  
  if (typeof module === "object" && module.exports) {
    module.exports = solveBeamBalanceAndMasses;
  }
  

File: CODIGO 139/9.kt
fun solveBeamBalanceAndMasses(test_cases: List<Int>): List<Int> {
    val masses = listOf(1, 3, 9, 27, 81)
    val result = mutableListOf<Int>()
  
    for (test_case in test_cases) {
      var powerOf3 = 0
      while (test_case >= 81) {
        powerOf3++
        test_case -= 81
      }
      result.add(masses[powerOf3])
      test_case -= masses[powerOf3]
    }
  
    return result
  }
  